uber/models/admin.py

```diff
@@ -0,0 +1,241 @@
+from datetime import datetime, timedelta
+
+import cherrypy
+from pytz import UTC
+from sideboard.lib import cached_property
+from sideboard.lib.sa import CoerceUTF8 as UnicodeText, UTCDateTime, UUID
+from sqlalchemy.orm import backref
+from sqlalchemy.schema import ForeignKey, UniqueConstraint
+from sqlalchemy.types import Boolean, Date, Float, Integer
+
+from uber.config import c
+from uber.decorators import presave_adjustment
+from uber.models import MagModel
+from uber.models.core import Attendee
+from uber.models.types import default_relationship as relationship, utcnow, \
+    Choice, DefaultColumn as Column, MultiChoice
+
+
+__all__ = [
+    'AdminAccount', 'PasswordReset', 'WatchList', 'DeptChecklistItem',
+    'Job', 'Shift']
+
+
+class AdminAccount(MagModel):
+    attendee_id = Column(UUID, ForeignKey('attendee.id'), unique=True)
+    hashed = Column(UnicodeText)
+    access = Column(MultiChoice(c.ACCESS_OPTS))
+
+    password_reset = relationship(
+        'PasswordReset', backref='admin_account', uselist=False)
+
+    def __repr__(self):
+        return '<{}>'.format(self.attendee.full_name)
+
+    @staticmethod
+    def is_nick():
+        return AdminAccount.admin_name() in c.JERKS
+
+    @staticmethod
+    def admin_name():
+        try:
+            from uber.models import Session
+            with Session() as session:
+                return session.admin_attendee().full_name
+        except:
+            return None
+
+    @staticmethod
+    def admin_email():
+        try:
+            from uber.models import Session
+            with Session() as session:
+                return session.admin_attendee().email
+        except:
+            return None
+
+    @staticmethod
+    def access_set(id=None):
+        try:
+            from uber.models import Session
+            with Session() as session:
+                id = id or cherrypy.session['account_id']
+                return set(session.admin_account(id).access_ints)
+        except:
+            return set()
+
+
+class PasswordReset(MagModel):
+    account_id = Column(UUID, ForeignKey('admin_account.id'), unique=True)
+    generated = Column(UTCDateTime, server_default=utcnow())
+    hashed = Column(UnicodeText)
+
+    @property
+    def is_expired(self):
+        return self.generated < datetime.now(UTC) - timedelta(days=7)
+
+
+class WatchList(MagModel):
```

Comment:
I think of this as registration/people-oriented, not administration oriented. Not exactly sure where it should go, since the core.py file is already huge... just wanted to note that I was surprised to see this here.